fritz

```diff
@@ -182,19 +182,28 @@ def patch_skyportal():
 
     # add Fritz-specific dependencies for SP
     # js
-    with open("extensions/skyportal/package.fritz.json", "r") as fpjf, open("skyportal/package.json", "r") as fpj:
-        pjf = json.load(fpjf)
-        pj = json.load(fpj)
-    pj["dependencies"] = {**pj["dependencies"], **pjf["dependencies"]}
-    with open("skyportal/package.json", "w") as fpj:
-        json.dump(pj, fpj, indent=2)
+    with open("extensions/skyportal/package.fritz.json", "r") as f:
+        fritz_pkg = json.load(f)
+    with open("skyportal/package.json", "r") as f:
+        skyportal_pkg = json.load(f)
+
+    skyportal_pkg["dependencies"] = {
+        **skyportal_pkg["dependencies"],
+        **fritz_pkg["dependencies"]
+    }
+    with open("skyportal/package.json", "w") as f:
+        json.dump(skyportal_pkg, f, indent=2)
+
     # python
-    with open(".requirements/ext.txt", "r") as frf, open("skyportal/requirements.txt", "r") as fr:
-        rf = frf.readlines()
-        r = fr.readlines()
-    with open("skyportal/requirements.txt", "w") as fr:
-        for line in rf + r:
-            fr.write(line)
+    with open(".requirements/ext.txt", "r") as f:
+        ext_req = f.readlines()
+    with open("skyportal/requirements.txt", "r") as f:
+        skyportal_req = f.readlines()
+    with open("skyportal/requirements.txt", "w") as f:
+        f.writelines(skyportal_req)
+        for line in ext_req:
+            if line not in skyportal_req:
+                f.write(line)
```

Comment:
Not sure if this will break anything, but what would happen if `ext_req` and `skyportal_req` had lines requiring the same package but different versions, for example? That is, if a package appears twice in the requirements.txt with different versions tagged how does pip resolve that? Do you know?

Reply:
pip will probably get very unhappy about that.

Reply:
(With the new resolver that they just released.)

Reply:
If we want to override version, we'd have to first parse the requirements and compare.  That's outside the scope of this PR, but something we could potentially add.

Reply:
Cool - and that will in turn get propagated as an error at some step when trying to stand up Fritz for the dev to resolve, right?

Reply:
Right yeah I think this is fine for now